cmake/ngraph.cmake

```diff
@@ -0,0 +1,62 @@
+include(ExternalProject)
+
+add_library(ngraph_interface_lib INTERFACE)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-comment")
+
+set(NGRAPH_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/3rdparty/ngraph)
+set(NGRAPH_SOURCE_DIR ${CMAKE_SOURCE_DIR}/3rdparty/ngraph)
+
+set(NGRAPH_EXTRA_CMAKE_FLAGS "")
+list(APPEND NGRAPH_EXTRA_CMAKE_FLAGS "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+list(APPEND NGRAPH_EXTRA_CMAKE_FLAGS "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}")
+list(APPEND NGRAPH_EXTRA_CMAKE_FLAGS "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
+list(APPEND NGRAPH_EXTRA_CMAKE_FLAGS "-DCMAKE_INSTALL_PREFIX=${NGRAPH_INSTALL_PREFIX}")
+list(APPEND NGRAPH_EXTRA_CMAKE_FLAGS "-DNGRAPH_USE_PREBUILT_LLVM=1")
+
+if(USE_NGRAPH_IE)
+  add_definitions(-DMXNET_USE_NGRAPH_IE=1)
+endif(USE_NGRAPH_IE)
+
+if(USE_NGRAPH_DISTRIBUTED)
+  find_package(MPI REQUIRED)
+  include_directories(SYSTEM ${MPI_C_INCLUDE_PATH} ${MPI_CXX_INCLUDE_PATH})
+  link_directories(${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES})
+  add_definitions(-DMXNET_USE_NGRAPH_DISTRIBUTED=1)
+  list(APPEND NGRAPH_EXTRA_CMAKE_FLAGS "-DNGRAPH_DISTRIBUTED_ENABLE=1")
+endif(USE_NGRAPH_DISTRIBUTED)
+
+message("${NGRAPH_EXTRA_CMAKE_FLAGS}")
+ExternalProject_Add(
+	ext_ngraph
+	SOURCE_DIR ${NGRAPH_SOURCE_DIR}
+
+	PREFIX 3rdparty/ngraph/
+	CMAKE_ARGS "${NGRAPH_EXTRA_CMAKE_FLAGS}"
+
+	BUILD_BYPRODUCTS ${NGRAPH_CMAKE_PREFIX}
+	BUILD_ALWAYS 1
+)
+
+add_dependencies(ngraph_interface_lib ext_ngraph)
+
+include_directories(
+  ${NGRAPH_INSTALL_PREFIX}/src/ext_ngraph-build/mkldnn/src/external/mkl/include
+)
+
+ExternalProject_Get_Property(ext_ngraph SOURCE_DIR INSTALL_DIR)
+
+target_include_directories(ngraph_interface_lib SYSTEM INTERFACE
+	${NGRAPH_INSTALL_PREFIX}/include
```

Comment:
Are we sure it makes sense to use `SYSTEM` here?  
- pro: `SYSTEM` will suppress warnings specific to nGraph header files.
- con: some of those ^ warnings might matter.
- con: adds to situations where `make` will fail to rebuild everything necessary.

Reply:
Will change